x-pack/plugins/cases/public/components/case_view/components/custom_fields.test.tsx

```diff
@@ -121,37 +121,39 @@ describe('Case View Page files tab', () => {
     ).not.toBeInTheDocument();
   });
 
-  it('adds missing custom fields with no custom fields in the case', async () => {
+  it('removes extra custom fields', async () => {
     appMockRender.render(
       <CustomFields
         isLoading={false}
-        customFields={[]}
-        customFieldsConfiguration={customFieldsConfigurationMock}
+        customFields={customFieldsMock}
+        customFieldsConfiguration={[
+          {
+            type: CustomFieldTypes.TOGGLE,
+            key: 'test_key_2',
+            label: 'My test label 2',
+            required: false,
+          },
+        ]}
         onSubmit={onSubmit}
       />
     );
 
-    userEvent.click((await screen.findAllByRole('switch'))[0]);
+    userEvent.click(await screen.findByRole('switch'));
 
     await waitFor(() => {
-      expect(onSubmit).toBeCalledWith([
-        {
-          type: CustomFieldTypes.TEXT,
-          key: 'test_key_1',
-          value: customFieldsConfigurationMock[0].defaultValue,
-        },
-        { type: CustomFieldTypes.TOGGLE, key: 'test_key_2', value: true },
-        customFieldsMock[2],
-        customFieldsMock[3],
-      ]);
+      expect(onSubmit).toBeCalledWith({
+        type: CustomFieldTypes.TOGGLE,
+        key: 'test_key_2',
+        value: false,
+      });
     });
   });
 
-  it('adds missing custom fields with some custom fields in the case', async () => {
+  it('replaces an existing toggle field correctly', async () => {
```

Comment:
I'm not sure I understand what `replaces` means here.

Maybe?
```suggestion
  it('clicks and submits an existing toggle field correctly', async () => {
```

Reply:
After reviewing the rest of the PR I know understand better why the keyword `replace` :D 

I still think here something like updates makes sense but it is no big deal so feel free to leave it as is!